File: internal/service/user/delete.go
package user

import (
	"context"
	"fmt"

	"github.com/google/uuid"
)

func (s *UserService) Delete(ctx context.Context, userID uuid.UUID) error {
	if err := s.DB.Q.DeleteUser(ctx, userID); err != nil {
		return fmt.Errorf("failed to delete user db: %w", err)
	}
	return nil
}

File: src/scripts/app/handlers/user/delete.ts
import { dropTables } from '@/src/scripts/database/handlers/drop';
import { deleteDataStore } from '@/src/scripts/prepareData';
import { router } from 'expo-router';

type DeleteUser = () => Promise<void>;

export const deleteUser: DeleteUser = async () => {
  try {
    await dropTables();
    await deleteDataStore();
    router.navigate('/');
  } catch (err) {
    console.error('Failed user delete:', err);
  }
};
